src/main/java/com/internship/backend/service/ReservationService.java

```diff
@@ -34,14 +33,18 @@ public class ReservationService {
     @Autowired
     private PriceRepository priceRepository;
 
+    @Autowired
+    private ReservationMapper reservationMapper;
+
     public List<Reservation> getAllReservations(){
         return reservationRepository.findAll();
     }
 
-    public Reservation addReservation(Reservation reservation) throws ReservationAlreadyExists, InvalidDateException {
-
+    public Reservation addReservation(ReservationDTO reservationDTO) throws ReservationAlreadyExists, InvalidDateException, TennisCourtDoesNotExistsException, UserDoesNotExistException {
+        Reservation reservation = reservationMapper.reservatonMapper(reservationDTO);
```

Comment:
As the method name is the same with the object one, it's not intuitive what should do. Could we find a better name for the method?